test/type_construction_test.rb

```diff
@@ -2983,6 +2983,75 @@ def test_if_annotation_success
     end
   end
 
+  def test_unreachable_if_void
```

Comment:
Please move these tests to `type_check_test.rb` if you can.

Reply:
Updated.